<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Course website for Biol 520C: Statistical modelling for biological data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Biol 520C Statistical modelling for biological data &mdash; UBC-Okanagan</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Lectures.html">Lectures</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Practicals
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_R_Introduction.html">Practical 01</a>
    </li>
    <li>
      <a href="02_Regression.html">Practical 02</a>
    </li>
    <li>
      <a href="03_MLR_Residuals.html">Practical 03</a>
    </li>
    <li>
      <a href="04_Visualisation.html">Practical 04</a>
    </li>
    <li>
      <a href="05_MEM_Model_Selection.html">Practical 05</a>
    </li>
    <li>
      <a href="06_Heteroskedasticity_Autocorrelation.html">Practical 06</a>
    </li>
    <li>
      <a href="07_Spatial_and_Phylogenetic_Autocorrelation.html">Practical 07</a>
    </li>
    <li>
      <a href="08_GLMs_for_Count_Data.html">Practical 08</a>
    </li>
    <li>
      <a href="09_Logistic_Regression.html">Practical 09</a>
    </li>
    <li>
      <a href="10_Stochastic_Simulation.html">Practical 10</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Practical_Datasets.html">Practicals</a>
    </li>
    <li>
      <a href="Paper_Datasets.html">Papers</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    R Help
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Plotting.html">Plotting</a>
    </li>
    <li>
      <a href="for_loops.html">for loops</a>
    </li>
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Course website for Biol 520C: Statistical
modelling for biological data</h1>

</div>


<div id="course-description" class="section level2">
<h2>Course Description</h2>
<p>Advanced course on biostatistics focused on combining data with
models to generate mechanistic descriptions of biological patterns.
Building from simple linear regression, the course teaches regression
methods for handling the most routinely encountered features in
biological data. Emphasis is placed on understanding and interpreting
the analyses in biological terms and linking analyses to theoretical or
applied questions. The course is divided into core lectures and
practicals.</p>
</div>
<div id="course-objectives" class="section level2">
<h2>Course Objectives</h2>
<p>This course combines lectures on theory and concepts with time
practicing statistical tools in R based software packages. The course is
designed to equip students with the tools and knowledge they need to
perform a variety of regression-based analyses that can model common
features of biological data. For each topic, there will be a core
lecture module and a lab based practical module. With the lectures,
students will learn the theory behind core modelling tools including
multiple linear regression, mixed effects models, generalised linear
models, and generalised least squares, while the practicals are intended
to provide a pathway that students can follow over the longer term as
their skills develop.</p>
<p>Learning Outcomes:</p>
<ul>
<li>Fit regression models using the principle of maximum likelihood</li>
<li>Analyze and interpret residuals from regression models to assess
model fit, and determine model improvement strategies. Fit and interpret
models that include both fixed and random effects.</li>
<li>Apply regression techniques to fit models with non-Gaussian error
distributions.</li>
<li>Fit and assess non-linear models, and interpret model
parameters.</li>
<li>Identify and manage non-independent data structures, including those
influenced by phylogenetic inertia and temporal/spatial
autocorrelation.</li>
<li>Apply AIC-based model selection to compare and evaluate competing
statistical models.</li>
<li>Build and implement data simulation models and conduct power
analyses.</li>
<li>Integrate research questions, study design, and analytical
approaches</li>
</ul>
</div>
<div id="course-format" class="section level2">
<h2>Course Format</h2>
<p><strong>Lectures</strong>: Lectures will cover the core concepts of
the course. Lecture slides will be posted on the course website the
evening prior to the lecture. Students are encouraged to take notes, and
to ask questions in the lectures. Lectures will be given in Zoom and a
link to the lecture is available here (Links to an external site.) and
in the Canvas navigation sidebar. All lectures will be recorded and made
available to the students. Calculators will be useful in the lectures
but are not mandatory. Questions will be used throughout the lectures to
reinforce key concepts. No grades are assigned to these questions, but
and students are encouraged to answer these to the best of their ability
in order to assist them in tracking their progress.</p>
<p><strong>Practicals</strong>: The practicals use structured tutorials
to guide students on the use of the open-source software program R for
applying the methods learned in the lectures to data. The lectures will
cover the material required for students to complete the practical
assignments, however, they are designed to be complementary and not all
the material in the practicals will be covered in the lectures and vice
versa. These also provide training on communicating the results of
statistical analyses, with a focus on reproducibility. Assessment of
student learning will be based on submitted assignments (see detailed
schedule below).</p>
<p><strong>Core project</strong>: The core research project is a major
component of the course and is designed to train students in data
analysis, interpretation, and scientific communication using best
practices in open science. The project is divided into two components
that mirror a typical analytical workflow: 1) Development of an initial
hypothesis and expected outcome(s), and 2) A verbal presentation of the
analyses and findings.</p>
</div>
<div id="pre-course-checklist" class="section level2">
<h2>Pre-course Checklist</h2>
<ul>
<li>The material learned in most entry level statistics course provides
important background and context on the topics that will be covered in
this course. It is recommended that you re-familiarise yourself with
this material before the course (e.g., understand the differences
between means, medians, and modes, variances, standard deviations,
etc.).</li>
<li>To fully engage in all of the course material will require a basic
familiarity with R. Working through the book `R for Beginners’ by
Paradis (available at <a href="https://cran.r-project.org"
class="uri">https://cran.r-project.org</a> (Links to an external site.)
–&gt; ~Documentation/Contributed) before beginning the course is a good
idea if you want to improve with R or with coding. Alternatively, the
swirl R package (available at: <a
href="https://swirlstats.com/students.html"
class="uri">https://swirlstats.com/students.html</a> (Links to an
external site.)) teaches R programming and data science interactively,
at your own pace, and right in the R console.</li>
<li>If you intend to use your personal computer during the course, it is
a good idea to try and install R, and RStudio before the course.</li>
</ul>
</div>
<div id="optional-material" class="section level2">
<h2>Optional Material</h2>
<p>There is no textbook for this course, but if students are interested
in expanding their knowledge beyond what is covered in the course, the
following textbooks are recommended:</p>
<ul>
<li>Bolker, B. M. (2008). Ecological models and data in R. Princeton
University Press.</li>
<li>Hilborn R, Mangel M. The ecological detective: confronting models
with data. 1997. Princeton University Press. ISBN 978-0691034973.</li>
<li>Zuur, A., Ieno, E. N., Walker, N., Saveliev, A. A., &amp; Smith, G.
M. (2009). Mixed effects models and extensions in ecology with R.
Springer Science &amp; Business Media.</li>
</ul>
</div>
<div id="course-evaluation" class="section level2">
<h2>Course Evaluation</h2>
<p>The assignments and evaluation for this course are structured along
two distinct, but complimentary lines: i) guided practical assignments;
and ii) a core research project. The practical assignments are designed
to provide students with the opportunity to learn how to apply the
methods learned in the lectures to real data using the statistical
software R. The core research project is designed to train students in
data analysis and presentation using best practices in open science. The
core project is divided into 3 components aimed at replicating the
typical analytical process: 1) initial hypothesis and expected
outcome(s); 2) a verbal presentation of the analyses and findings; and
3) the submission of a paper describing the work.</p>
<table>
<colgroup>
<col width="42%" />
<col width="7%" />
<col width="50%" />
</colgroup>
<tbody>
<tr class="odd">
<td>Hypothesis and expected outcome(s)</td>
<td>5%</td>
<td>Week 4</td>
</tr>
<tr class="even">
<td>Presentation</td>
<td>30%</td>
<td>Weeks 12 &amp; 13</td>
</tr>
<tr class="odd">
<td>Practicals (10)</td>
<td>25%</td>
<td>Due on ~weekly basis (see schedule below)</td>
</tr>
<tr class="even">
<td>Participation from practicals</td>
<td>10%</td>
<td>Due on ~weekly basis (see schedule below)</td>
</tr>
<tr class="odd">
<td>Take-home exam</td>
<td>30%</td>
<td>Week 14</td>
</tr>
<tr class="even">
<td>Total</td>
<td>100%</td>
<td></td>
</tr>
</tbody>
</table>
<div id="practical-assignments-and-participation-35"
class="section level3">
<h3>Practical Assignments and Participation (35%)</h3>
<p>Beginning in the first week of class, students will complete
practical assignments on an approximately weekly basis (see schedule
below). A total of ten practical assignments will be completed
throughout the course. These practicals are designed to provide hands-on
experience applying the statistical methods introduced in lectures to
real biological datasets using the statistical software R. The primary
purpose of the practical assignments is to ensure that students develop
the practical skills required to conduct effective statistical analyses
and communicate results clearly and reproducibly. The skills developed
through these assignments are directly transferable to students’ thesis
work and independent research projects. Practical assignments, along
with the associated datasets, will be hosted on the course GitHub
webpage. All submissions must be completed electronically via Canvas.
Lectures will be held on Mondays and Wednesdays. By the end of the
Wednesday lecture, all material required to complete that week’s
practical will have been covered. Practical assignments are due before
the start of the following Wednesday lecture.</p>
<p><strong>Grading</strong>: Each practical assignment is worth 3.5% of
the final course grade, for a total of 35% across all ten practicals. Of
this:</p>
<ul>
<li><p>1% is awarded for submitting the assignment on time, regardless
of correctness. This component constitutes the participation grade for
the course.</p></li>
<li><p>The remaining 2.5% is based on the accuracy of the responses, the
quality and clarity of the analysis, well-structured and well-commented
code, and the quality of figures and data presentation.</p></li>
</ul>
<p>Solutions to each practical assignment will be posted after the
submission deadline. Late practicals will be accepted after solutions
have been released; however, late submissions will not receive the
participation component of the grade. No extensions will be granted for
practical assignments except under documented exceptional
circumstances.</p>
</div>
<div id="hypothesis-and-expected-outcomes-5" class="section level3">
<h3>Hypothesis and expected outcome(s) (5%)</h3>
<p>By the end of Week 4, students must submit a one-page proposal
describing the study system they will use for their course project. The
proposal should clearly state the initial hypothesis, the expected
outcome(s), and the dataset(s) that will be used to address the research
question. This assignment is intended to function as a light form of
preregistration. Its purpose is to encourage students to articulate
their analytical questions, hypotheses, and expectations a priori,
before conducting formal analyses. By doing so, this assignment helps
distinguish between confirmatory analyses (testing pre-specified
hypotheses) and exploratory analyses, and reduces the risk of post hoc
hypothesis generation, fishing expeditions, or p-hacking. Students are
not penalised if the results ultimately do not support the initial
hypothesis. Instead, the emphasis is on clearly documenting the original
expectations and analytical intent prior to seeing the results.</p>
<p><strong>Grading</strong>: This assignment is assessed on a completion
basis. All proposals that meet the submission requirements will receive
full credit (5/5). Late submissions will receive a grade of 0.
Submission of this assignment is a prerequisite for all subsequent core
project components. No further project-related assignments will be
accepted until the hypothesis and expected outcome(s) proposal has been
submitted.</p>
</div>
<div id="presentations-30" class="section level3">
<h3>Presentations (30%)</h3>
<p>Students will complete a core data analysis assignment as the
capstone component of the course. Each student will deliver a 15-minute
conference-style presentation based on their analyses. This assignment
is designed to provide students with an opportunity to apply the
statistical modelling tools introduced in lectures and practicals in an
unguided setting, analogous to analysing data outside of a classroom
environment. For this presentation, students will apply the modelling
approaches covered in the course to an empirical dataset of their
choosing (see Datasets below) and present their work in a professional
scientific format. Presentations should be organised into the following
sections:</p>
<p><strong>Presentation structure</strong></p>
<p><strong>Introduction</strong> Provide a brief description of the
study system and clearly state the research question(s) being addressed.
Relevant background literature should be cited. Recommended length: ~2–3
minutes.</p>
<p><strong>Methods</strong> Briefly describe how the data were collected
and what variables are included. Provide a detailed description of the
analytical workflow applied to the data, including model structure,
assumptions, and justification for methodological choices. All relevant
literature and statistical packages should be cited. The description
should be sufficiently detailed that the analysis could be reproduced by
others with access to the data. Recommended length: ~6–7 minutes.</p>
<p><strong>Results</strong> Present the statistical results using
appropriate figures and tables. Results should be explicitly contrasted
with the predictions stated in the Hypothesis and Expected Outcome(s)
assignment, noting where outcomes align with or differ from initial
expectations. Students are not evaluated on whether results support
their initial hypotheses, but on the clarity, appropriateness, and
transparency of their analytical decisions and interpretation.
Recommended length: ~3–5 minutes.</p>
<p><strong>Discussion / Conclusion</strong> Summarise the key findings
and place them in a biological context. Unexpected or null results
should be discussed where relevant. Recommended length: ~1–2
minutes.</p>
<p><strong>References</strong> Include references for all data sources,
methods, and background literature.</p>
<p><em>Note: The recommended section lengths are guidelines rather than
strict requirements. The primary emphasis of the presentation should be
on the statistical modelling itself, including rationale for analytical
choices, assumptions of the methods used, and potential sources of bias
or uncertainty. Overly long introductions or discussions should be
avoided.</em></p>
<p><strong>Reproducibility appendix</strong></p>
<p>In addition to the in-class presentation, students must submit a
reproducibility appendix generated using R Markdown (PDF, HTML, or Word
format). This document should detail every step of the analysis
presented, including all data processing, model fitting, and figure
generation. Code should be clearly commented and structured to allow
full reproducibility of the results shown in the presentation.</p>
<p><strong>Datasets</strong>: Students may select from a set of
pre-approved datasets provided on the course website. Students are
strongly encouraged to use their own data where possible; however,
instructor approval is required. In general, there are no restrictions
on data type, provided that the statistical methods covered in the
course are appropriate. Datasets requiring specialised bioinformatics or
preprocessing pipelines (e.g., RNA-seq pipelines) are not permitted.
Students intending to use their own data are encouraged to discuss
suitability with the instructor as early as possible.</p>
<p><strong>Submission:</strong> Presentation slides must be submitted in
advance of the presentation date (exact deadline to be announced later
in the semester). The order of presentations will be determined randomly
in class on the day of the presentations.</p>
<p><strong>Grading:</strong> The presentation will be worth 20% of your
total grade. Grading will based on a pre-provided rubric. The appendix
material will be worth 10% of your total grade.</p>
</div>
<div id="take-home-final-exam-30" class="section level3">
<h3>Take-Home Final Exam (30%)</h3>
<p>The take-home final exam will consist of a set of questions designed
to assess students’ understanding of the core statistical concepts
covered in the course. The exam will be comprehensive, and material from
both lectures and practical assignments may be tested. The exam will
emphasise statistical reasoning, interpretation, and model evaluation,
rather than rote calculation. Questions may require students to
interpret model outputs, assess assumptions, justify analytical choices,
and evaluate alternative modelling approaches.</p>
<p><strong>Grading:</strong> The final exam is worth 30% of the final
course grade.</p>
</div>
</div>
<div id="lecture-outline" class="section level2">
<h2>Lecture Outline</h2>
<p>(Approximate schedule of topics covered in lectures)</p>
<table>
<colgroup>
<col width="8%" />
<col width="41%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Week</th>
<th>Lecture Topics</th>
<th>Practical Assignment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Course introduction; Regression refresher</td>
<td>Practical 01: Introduction to R as a programming language</td>
</tr>
<tr class="even">
<td>2</td>
<td>Probability theory; Likelihood; Maximum likelihood</td>
<td>Practical 02: Deterministic and stochastic models, simple linear
regression, and maximum likelihood</td>
</tr>
<tr class="odd">
<td>3</td>
<td>Multiple linear regression; Parameter interactions; Interpreting
residuals</td>
<td>Practical 03: Multiple linear regression, Residuals</td>
</tr>
<tr class="even">
<td>4</td>
<td>Mixed effects models; Model Selection; Information criterion</td>
<td>Practical 04: Data visualisation; Hypothesis and predictions
due</td>
</tr>
<tr class="odd">
<td>5</td>
<td>Model Selection; Model averaging; Heteroskedasticity</td>
<td>Practical 05: Mixed effect models, model selection, and model
averaging</td>
</tr>
<tr class="even">
<td>6</td>
<td>Temporal autocorrelation; Spatial Autocorrelation</td>
<td>Practical 06: Heteroskedasticity; Autocorrelation Function; Models
with autocorrelated errors</td>
</tr>
<tr class="odd">
<td>7</td>
<td>Phylogenetic inertia; Phylogenetically controlled regression</td>
<td>Practical 07: Spatial and Phylogenetic Autocorrelation</td>
</tr>
<tr class="even">
<td>8</td>
<td>Non-Gaussian residuals; Generalised linear regression Poisson and
Logistic regression</td>
<td>Practical 08: Generalised Linear Models for Count Data</td>
</tr>
<tr class="odd">
<td>9</td>
<td>Zero-inflated data; Non-linear modelling; Deterministic
functions</td>
<td>Practical 09: GLMs for Binary and Proportion Data</td>
</tr>
<tr class="even">
<td>10</td>
<td>Stochastic simulation and power analysis; Course Overview</td>
<td>Practical 10: Stochastic Simulations, Power Analysis, Non-linear
models</td>
</tr>
<tr class="odd">
<td>11</td>
<td>Independent Project Work</td>
<td>No Practical</td>
</tr>
<tr class="even">
<td>12</td>
<td>Student presentations</td>
<td>No Practical</td>
</tr>
<tr class="odd">
<td>13</td>
<td>Student presentations</td>
<td>Term presentations due (35%)</td>
</tr>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
